source/includes/_wallet.md

```diff
@@ -1 +1,1029 @@
+    "keys": []
+  }
+}
+```
+
+Bcoin maintains a wallet database which contains every wallet. Wallets are not usable without also using a wallet database. For testing, the wallet database can be in-memory, but it must be there.
+
+Wallets in bcoin use bip44. They also originally supported bip45 for multisig, but support was removed to reduce code complexity, and also because bip45 doesn't seem to add any benefit in practice.
+
+The wallet database can contain many different wallets, with many different accounts, with many different addresses for each account. Bcoin should theoretically be able to scale to hundreds of thousands of wallets/accounts/addresses.
+
+Each account can be of a different type. You could have a pubkeyhash account, as well as a multisig account, a witness pubkeyhash account, etc.
+
+Note that accounts should not be accessed directly from the public API. They do not have locks which can lead to race conditions during writes.
+
+## Wallet Options
+> Wallet options object will look something like this
+
+```json
+  {
+    "id": "walletId",
+    "witness": true,
+    "watchOnly": false,
+    "accountKey": key,
+    "accountIndex": 1,
+    "type": "pubkeyhash"
+    "m": 1, 
+    "n": 1,
+    "keys": [],
+    "mnemonic": 'differ trigger sight sun undo fine sheriff mountain prison remove fantasy arm'
+  }
+```
+Options are used for wallet creation. None are required. 
+
+### Options Object
+Name | Type | Default | Description
+---------- | ----------- | -------------- | -------------
+id | String |  | Wallet ID (used for storage)
+master | HDPrivateKey/HDPublicKey | | | Master HD key. If not present, it will be generated
+witness | Boolean | `false` | Whether to use witness programs
+accountIndex | Number | `0` | The BIP44 [account index](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#Account)
+receiveDepth | Number | | The index of the _next_ receiveing address
+changeDepth | Number | | The index of the _next_ change address
+type | String | `'pubkeyhash'` |Type of wallet (pubkeyhash, multisig)
+compressed | Boolean | `true` | Whether to use compressed public keys
+m | Number | `1` | m value for multisig (`m-of-n`)
+n | Number | `1` | n value for multisig (`m-of-n`)
+mnemonic | String | | A mnemonic phrase to use to instantiate an hd private key. One will be generated if none provided
+
+
+## Wallet Auth
+> The following samples return a wallet object
+
+```shell--vars
+    token='977fbb8d212a1e78c7ce9dfda4ff3d7cc8bcd20c4ccf85d2c9c84bbef6c88b3c'
+    id='primary'
+```
+
+```shell--curl
+curl $url/wallet/$id/send \
+    -H 'Content-Type: application/json' \
+    -d '{ "token": "$token" ... }'
+```
+
+```shell--cli
+bcoin cli wallet get --network=testnet --token=$token
+```
+
+```javascript
+`use strict`
+
+const httpWallet = new bcoin.http.Wallet({ 
+    id: id, 
+    token: token,
+    network: 'testnet',
+    db: 'leveldb'
```

Comment:
Good call. I stopped adding most of the options since a lot can be set as globals or aren't necessarily needed. Hopefully it's not too confusing though.